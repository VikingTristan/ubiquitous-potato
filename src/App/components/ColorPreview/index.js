import React from "react";
import PropTypes from "prop-types";

const ColorPreview = ({ name, hex, rgb, cmyk }) => (
    <figure>
        <svg className={name === "White" ? "border" : null}>
            <rect width="100%" height="100%" fill={hex} />
        </svg>
        <figcaption>
            {name}<br/>
            <small>Hex color code: {hex}</small>
            {rgb && <><br/><small>RGB color code: {rgb}</small></> }
            {cmyk && <><br/><small>CMYK color code: {cmyk}</small></>}
        </figcaption>
    </figure>
);

ColorPreview.propTypes = {
    name: PropTypes.string.isRequired,
    hex: PropTypes.string.isRequired,
    rgb: PropTypes.string,
    cmyk: PropTypes.string
};

export default ColorPreview;
