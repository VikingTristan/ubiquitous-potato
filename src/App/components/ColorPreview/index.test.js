import React from "react";
import { shallow } from "enzyme";

import ColorPreview from "./index";

describe("Component: ColorPreview", () => {
    it("is defined", () => {
        expect(ColorPreview).toBeDefined();
    });

    it("renders, without RGB and CMYK when not provided", () => {
        const wrapper = shallow(<ColorPreview name="test" hex="#test" />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(<small>Hex color code: #test</small>)).toBeTruthy();
        expect(wrapper.contains(<small>RGB color code: test,test,test</small>)).toBeFalsy();
        expect(wrapper.contains(<small>CMYK color code: test/test/test/test</small>)).toBeFalsy();
        expect(wrapper.html()).toContain("fill=\"#test\"");
    });

    it("renders, with RGB and CMYK when provided", () => {
        const mockElement = {
            name: "test",
            hex: "#test",
            rgb: "test,test,test",
            cmyk: "test/test/test/test"
        };
        const wrapper = shallow(<ColorPreview {...mockElement} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(<small>Hex color code: #test</small>)).toBeTruthy();
        expect(wrapper.contains(<small>RGB color code: test,test,test</small>)).toBeTruthy();
        expect(wrapper.contains(<small>CMYK color code: test/test/test/test</small>)).toBeTruthy();
        expect(wrapper.html()).toContain("fill=\"#test\"");
    });

});
