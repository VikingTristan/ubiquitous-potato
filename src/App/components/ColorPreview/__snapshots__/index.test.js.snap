// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ColorPreview renders, with RGB and CMYK when provided 1`] = `
<figure>
  <svg
    className={null}
  >
    <rect
      fill="#test"
      height="100%"
      width="100%"
    />
  </svg>
  <figcaption>
    test
    <br />
    <small>
      Hex color code: 
      #test
    </small>
    <br />
    <small>
      RGB color code: 
      test,test,test
    </small>
    <br />
    <small>
      CMYK color code: 
      test/test/test/test
    </small>
  </figcaption>
</figure>
`;

exports[`Component: ColorPreview renders, without RGB and CMYK when not provided 1`] = `
<figure>
  <svg
    className={null}
  >
    <rect
      fill="#test"
      height="100%"
      width="100%"
    />
  </svg>
  <figcaption>
    test
    <br />
    <small>
      Hex color code: 
      #test
    </small>
  </figcaption>
</figure>
`;
